src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -67,12 +69,13 @@ def __init__(self) -> None:
         self.federation_options: dict[int, ConfigsRecord] = {}
         self.task_ins_store: dict[UUID, TaskIns] = {}
         self.task_res_store: dict[UUID, TaskRes] = {}
+        self.task_ins_id_to_task_res: dict[UUID, TaskRes] = {}
```

Comment:
We already store each `TaskRes` in `task_res_store`, should we perhaps just store a mapping from `TaskIns` ID to `TaskRes` ID?

```suggestion
        self.task_ins_id_to_task_res: dict[UUID, UUID] = {}
```

Reply:
Sure thing, I've updated the PR.